src/Reader/Iterable/Processor/Like.php

```
@@ -15,6 +15,9 @@ public function getOperator(): string
 
     public function match(array $item, array $arguments, array $filterProcessors): bool
     {
+        if(count($arguments) < 2) {
+            throw new \RuntimeException('The count of arguments is too small!');
```

```suggestion
            throw new \RuntimeException('At least two arguments should be provided');
```

---

AFAIK it should be `Exactly two arguments`, since any additional elements in `$arguments` will be silently ignored. I would rather suggest something like `$arguments should contain exactly two elements` or ignore this check completely, since there is so many cases when this could fail (associative array, non-scalars).

---

@rob006 if we were to require the array to contain the indexes `0` and `1`? I do not consider filtering exactly to 2 items as it may adversely affect future changes.
Which could be a problem here, because one of the demands was that requests could be transfered between systems (example json). It is not required to run the same yii version on both systems.

---

I don't think that is valid reason for error messages not being precise.

---

@kamarton I suggest to address this in future, when real use case came up. Removing such constraint should be easy, it should not block any changes in future.

---

> I suggest to address this in future, when real use case came up. Removing such constraint should be easy, it should not block any changes in future.

Acceptable. Remove only this (and similar), or can the whole PR be closeable?

---

What was meant is that error message and check should be as https://github.com/yiisoft/data/pull/27#discussion_r334285603